cranelift/codegen/src/opts/algebraic.isle

```diff
@@ -39,6 +39,14 @@
 (rule (simplify (imul ty (ineg ty x) (ineg ty y)))
       (subsume (imul ty x y)))
 
+;; iabs(ineg(x)) == iabs(x).
+(rule (simplify (iabs ty (ineg ty x)))
+      (subsume (iabs ty x)))
+
+;; iabs(iabs(x)) == x.
+(rule (simplify (iabs ty inner @ (iabs ty x)))
+      (subsume inner))
+
```

Comment:
Is this true? If `x` is `-1` won't this rewrite to `-1` when `abs(abs -1)` will be `1`?

Reply:
Oh, reading the rule I think it's just the comment and PR description that are wrong. It's actually:
```
iabs (iabs(x)) == iabs(x)
```

Reply:
Good catch!